docs/API/V1/list.md

```diff
@@ -252,83 +259,93 @@ will return a JSON document with the keys `config`, `date`, `hostname_f`,
+                    "hosts": null,
+                    "trigger": null
+                },
+                "tools/dbutenho.bos.csb": {
+                    "hostname-alias": null,
+                    "hostname-all-fqdns": null,
+                    "hostname-all-ip-addresses": null,
+                    "hostname-domain": null,
+                    "hostname-fqdn": null,
+                    "hostname-ip-address": null,
+                    "hostname-nis": null,
+                    "hostname-short": null,
+                    "label": null,
+                    "rpm-version": null,
+                    "tools": null,
+                    "vmstat": null
+                },
+                "tools/dbutenho.bos.csb/vmstat": {
+                    "install_check_output": null,
+                    "install_check_status_code": null,
+                    "options": null
+                }
             },
-            "tools": {
-                "group": null,
-                "hosts": null,
-                "trigger": null
-            },
-            "tools/dbutenho.bos.csb": {
-                "hostname-alias": null,
-                "hostname-all-fqdns": null,
-                "hostname-all-ip-addresses": null,
-                "hostname-domain": null,
-                "hostname-fqdn": null,
-                "hostname-ip-address": null,
-                "hostname-nis": null,
-                "hostname-short": null,
-                "label": null,
-                "rpm-version": null,
-                "tools": null,
-                "vmstat": null
-            },
-            "tools/dbutenho.bos.csb/vmstat": {
-                "install_check_output": null,
-                "install_check_status_code": null,
-                "options": null
-            }
+            "name": null,
+            "owner_id": null,
+            "resource_id": null,
+            "uploaded": null
         },
-        "name": null,
-        "owner_id": null,
-        "resource_id": null,
-        "uploaded": null
-    },
-    "server": {
-        "deletion": null,
-        "index-map": {
-            "container-pbench.v6.run-data.2023-03": null,
-            "container-pbench.v6.run-toc.2023-03": null
-        },
-        "origin": null,
-        "tarball-path": null
+        "server": {
+            "deletion": null,
+            "index-map": {
+                "container-pbench.v6.run-data.2023-03": null,
+                "container-pbench.v6.run-toc.2023-03": null
+            },
+            "origin": null,
+            "tarball-path": null
+        }
     }
 }
 ```
+
+### Combining key namespace summary and date range
+
+When both the `keysummary` and `daterange` query parameters are `true`, the
+`application/json` response body contains the `from`, `to`, and `keys` key
+values. If the selected collection filters produce no results, as with
+`daterange` alone, the `from` and `to` keys will be omitted and the value of
+`keys` will be an empty object.
```

Comment:
Should we have verbiage in the "Key namespace summary" section, like we do in the "Dataset date range" section, which describes what is returned when no results are selected?

Er, but...doesn't [this line](https://github.com/distributed-system-analysis/pbench/blob/4e79e67fcc9cf4386d693aff1301c67a7f8d182a/lib/pbench/server/api/resources/datasets_list.py#L336) prevent the `keys` object from ever being empty?

(Either way, we should probably have a unit test which checks the documentation assertion here....)

Reply:
Yeah, you're right; and thinking about it now I'm not sure how I want to resolve that. We're always describing the `dataset` SQL table columns, which, arguably, we "shouldn't" be doing if there are no matches. However testing whether there are matches is a little messy since you can't really `if <iterator>` or `if len(<iterator>)`, and I'd rather not list-ify it... but the alternative is to do it on the first iteration of the `for`, which is (almost?) equally ugly if a bit more scaleable.